### PR TITLE
[Housekeeping] Refactor flytectl flags

### PR DESCRIPTION
**Pull request by aeioulisa**
# TL;DR
Flytectl code has one issue. Internal method/pkg use flags directly rather than using them as parameters. We should only use flags in the root command. After that, we should pass the value in the method.

## Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [X] Code completed
- [X] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
Just refactor.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1617

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pmahindrakar-oss** [comment]
The description and the changes don't seems to be on the same lines.

Can you clarify since I just see variable and import path renames

File: cmd/register/register_util.go [hunk 1]
```diff
@@ -215,15 +214,15 @@ func hydrateIdentifier(identifier *core.Identifier) {
...
+		identifier.Version = defaultFilesConfig.Version
 	}
 }
 
 func hydrateTaskSpec(task *admin.TaskSpec, sourceCode string) error {
 	if task.Template.GetContainer() != nil {
 		for k := range task.Template.GetContainer().Args {
 			if task.Template.GetContainer().Args[k] == "" || task.Template.GetContainer().Args[k] == registrationRemotePackagePattern {
-				remotePath, err := getRemoteStoragePath(context.Background(), Client, rconfig.DefaultFilesConfig.SourceUploadPath, sourceCode, rconfig.DefaultFilesConfig.Version)
+				remotePath, err := getRemoteStoragePath(context.Background(), Client, defaultFilesConfig.SourceUploadPath, sourceCode, defaultFilesConfig.Version)
```
Comment (pmahindrakar-oss) [inline cmd/register/register_util.go:225]: I think the main agenda was to not use such config directly in these internal functions and be used only the first level command function .
Rest of the places we want to pass the config as parameter wherever it's used

Comment (aeioulisa) [reply cmd/register/register_util.go]: @pmahindrakar-oss thanks for your review
I've been pushed a new commit
Could you please review it?

**Comment by aeioulisa** [comment]
@pmahindrakar-oss thanks for your review
I've been pushed a new commit
Could you please review it?

**Review by pmahindrakar-oss — APPROVED** [review]
Looks good to me. Thanks for making those changes .

**Review by evalsocket — APPROVED** [review]

**Pull request merged**